src/main/java/pl/edu/icm/pl/mxrdr/extension/importer/pdb/PdbXmlParser.java

```diff
@@ -0,0 +1,196 @@
+package pl.edu.icm.pl.mxrdr.extension.importer.pdb;
+
+import com.google.common.collect.Lists;
+import edu.harvard.iq.dataverse.importer.metadata.ResultField;
+import io.vavr.control.Try;
+import org.dom4j.Document;
+import org.dom4j.DocumentHelper;
+import org.dom4j.Node;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.edu.icm.pl.mxrdr.extension.importer.MxrdrMetadataField;
+import pl.edu.icm.pl.mxrdr.extension.importer.SymmetryStructureMapper;
+
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Singleton
+public class PdbXmlParser {
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    // -------------------- LOGIC --------------------
+
+    /**
+     * Parses the pdb xml to our metadata model and science team demands.
+     */
+    List<ResultField> parse(String pdbXml) {
+
+        Document document = Try.of(() -> DocumentHelper.parseText(pdbXml))
```

Comment:
If we know we can safely load the whole file into memory, consider using XML to Object mapping library like Jackson that is present on our classpath already instead of manual parsing. API client object might then return a list of objects that would later only have to be converted into our internal domain model. Those mapping objects might also encapsulate the extra logic of i.e. lists represented as `#` separated `String`s and such.

Reply:
About extra logic in our object. I feel like the actual value object should be data container, and the parser should contain business logic, what do you think?